uplink/ecclient/client.go

```diff
@@ -213,42 +213,35 @@ func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrd
 		)
 		return nil, err
 	}
-	defer func() {
-		if ctx.Err() != nil || err != nil {
-			hash = nil
-			err = errs.Combine(err, upload.Cancel(ctx))
-			return
-		}
-		h, closeErr := upload.Commit(ctx)
-		hash = h
-		err = errs.Combine(err, closeErr)
-	}()
 
 	_, err = sync2.Copy(ctx, upload, data)
 	// Canceled context means the piece upload was interrupted by user or due
 	// to slow connection. No error logging for this case.
-	if ctx.Err() == context.Canceled {
-		if parent.Err() == context.Canceled {
-			ec.log.Info("Upload to node canceled by user", zap.String("NodeID", storageNodeID.String()))
+	if err != nil {
```

Comment:
Will `context.Canceled` error be returned from line 217?

Reply:
It will.
Whatever error returned by `ctx` is caught by `sync2.Copy` and returned to the caller. See https://github.com/storj/storj/blob/9dbd621c604d0a5d4bf13878ee8e62268b9b15c8/internal/sync2/copy.go#L25